nexus/types/src/external_api/views.rs

```diff
@@ -285,14 +285,36 @@ pub struct Rack {
 
 // SLEDS
 
-/// Client view of an [`Sled`]
+/// Client view of a [`Sled`]
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct Sled {
     #[serde(flatten)]
     pub identity: AssetIdentityMetadata,
     pub service_address: SocketAddrV6,
 }
 
+// PHYSICAL DISKS
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum PhysicalDiskType {
+    Internal,
+    External,
+}
+
+/// Client view of a [`PhysicalDisk`]
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct PhysicalDisk {
+    #[serde(flatten)]
+    pub identity: AssetIdentityMetadata,
+
+    pub vendor: String,
+    pub serial: String,
+    pub model: String,
+
+    pub disk_type: PhysicalDiskType,
+}
```

Comment:
```suggestion
pub struct PhysicalDisk {
    #[serde(flatten)]
    pub identity: AssetIdentityMetadata,

    pub vendor: String,
    pub serial: String,
    pub model: String,

    pub disk_type: PhysicalDiskType,
    pub sled_id: Uuid,
}
```

Including `sled_id` here is helpful. We include the parent's ID on most (though not all) resources. We should probably work to unify that.

Reply:
Added, but as an `Option`, since I want to preserve the ability to tell users about disks which may no longer be attached to any sleds for historical purposes.